### DISCUSSION TITLE
Blog Post [4]

### DISCUSSION THREAD

**Discussion by gordi76** [NicoReimer/Breakpoint#10] [Weekly-Blog]
### Klassenmodellierung

In der vergangenen Woche haben wir uns mit der Struktur unseres Projekts auseinandergesetzt und das grundlegende Klassendiagramm entworfen. Dabei stand im Vordergrund, eine solide Basis zu schaffen, auf der wir im weiteren Projektverlauf aufbauen können. Das Klassendiagramm bildet die wichtigsten Komponenten unseres Systems ab und zeigt, wie die Kernklassen interagieren und welche Verantwortlichkeiten sie jeweils haben.

Unser Fokus lag darauf, die einzelnen Klassen so zu gestalten, dass sie klare Verantwortlichkeiten haben, um eine stabile Grundlage für spätere Erweiterungen zu legen. Besonders wichtig war uns, dass die Inhalte (`Textbox`, `Table`, `Image`) flexibel gehandhabt werden können. Hierfür haben wir eine abstrakte Content-Klasse eingeführt, die verschiedene Inhaltstypen erlaubt und zukünftige Erweiterungen erleichtert. Die Wiki- und Page-Klassen legen die Struktur für unser Wiki-System fest, während Benutzerrechte durch die Klassen User, Role, und Right verwaltet werden.

Beziehungen zwischen den Klassen wurden bewusst nur dort hinzugefügt, wo sie für das Verständnis der Grundstruktur erforderlich waren. Unsere Priorität war, zunächst die wesentlichen Funktionen darzustellen, ohne die Struktur unnötig zu verkomplizieren. So behalten wir die Flexibilität, Details später anzupassen, falls das System weiterentwickelt wird.

### SOLID-Prinzip

Wir haben das Single Responsibility Principle (SRP) aus den SOLID-Prinzipien bei der Gestaltung unseres Klassendiagramms berücksichtigt.

**Comment by Sebastian-Albert** [comment]
Hallo liebes Breakpoint-Team,
euer Blogeintrag habt ihr sehr detailliert und sehr ausführlich geschrieben. Das finde ich gut! Ich finde auch sehr gut, dass ihr das Klassendiagramm gut geplant habt und dass ihr euch sehr viele Gedanken über das Klassendiagramm gemacht habt. Euer Klassendiagramm finde ich sehr gelungen. Ich habe ein paar Verbesserungsvorschläge für euch: 1. In eurem Klassendiagramm sind keine Rollennamen eingetragen. 2. Euer Klassendiagramm besteht fast nur aus Compositionen und Aggregationen. Man benutzt Compositionen und Aggregationen eher selten. 3. Im Blogeintrag habt ihr euer Klassendiagramm nicht verlinkt und nicht erwähnt, wo sich das Klassendiagramm befindet. 4. Eure Ordnerstruktur könnt ihr noch verbessern. Weil in der Aufgabenstellung werden wir darauf hingewiesen, dass die Dokumentation klar vom Quellcode getrennt sein soll. Alles in allem finde ich euer Klassendiagramm aber sehr gelungen!
Viele Grüße
Luca, Stefan, Philip, Oskar und Sebastian Albert

**Comment by GermanJesus-lul** [comment]
Hey,

Euer Klassendiagramm sieht gut aus. Eure Beziehungen sind gut durchdacht und entsprechen den UML-Klassifizierungen. Mir gefällt auch die eingeplante Erweiterbarkeit eures Aufbaus. 

Aufgefallen ist mir nur, dass ihr teilweise "+" und teilweise "-" bei der Auflistung der Methoden/Attribute verwendet. Hat das einen Grund?

Mit freundlichen Grüßen,
Julius und das Rezeptbuch Team

**Reply by NicoReimer** [reply]
Hey, 

Die + und - Symbole zeigen ob die Funktion oder das Attribut public (+) oder private (-) ist

Gruß
Das Breakpoint Team
